### PR TITLE
feat(blocks): add options config in database

### PR DESCRIPTION
**Pull request by fundon**
* make database options configurable
* add `Copy link to block` button to options on AFFiNE side

### CODE REVIEW

**Comment by fundon** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/8022?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#8022** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/8022?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#8001** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/8001?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#8021](https://github.com/toeverything/blocksuite/pull/8021) <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/8021?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#7945** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/7945?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#7953](https://github.com/toeverything/blocksuite/pull/7953) <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/7953?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#7944** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/7944?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#7964** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/7964?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @fundon and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/blocks/src/database-block/database-block.ts [hunk 1]
```diff
@@ -59,58 +60,58 @@ export class DatabaseBlockComponent extends CaptionedBlockComponent<
...
-                class: 'delete-item',
-                name: 'Delete Database',
-                select: () => {
-                  this.model.children.slice().forEach(block => {
-                    this.doc.deleteBlock(block);
-                  });
-                  this.doc.deleteBlock(this.model);
-                },
+        },
+        // {
```
Comment (Saul-Mirone) [inline packages/blocks/src/database-block/database-block.ts:86]: comment

**Review by zzj3720 — APPROVED** [review]

File: packages/affine/components/src/rich-text/inline/presets/affine-inline-specs.ts [hunk 2]
```diff
@@ -16,10 +18,11 @@ export interface AffineTextAttributes {
...
   code?: true | null;
   link?: string | null;
-  reference?: {
-    type: 'Subpage' | 'LinkedPage';
-    pageId: string;
-  } | null;
+  reference?:
+    | ({
+        type: 'Subpage' | 'LinkedPage';
+      } & ReferenceInfo)
```
Comment (Saul-Mirone) [inline packages/affine/components/src/rich-text/inline/presets/affine-inline-specs.ts:24]: @Flrande PTAL

Comment (Flrande) [reply packages/affine/components/src/rich-text/inline/presets/affine-inline-specs.ts:24]: LGTM

**Review by Saul-Mirone — APPROVED** [review]

**Pull request merged**